tests/ops/op_math/test_controlled.py

```diff
@@ -1151,14 +1152,6 @@ def test_decomposition_undefined(self):
         with pytest.raises(DecompositionUndefinedError):
             op.decomposition()
 
-    def test_global_phase_decomp_raises_warning(self):
```

Comment:
What happens now? Are we getting an explicit error now? So maybe we should update this test to catch that error instead.

Reply:
I would think this warning should still be raised no? Since the decomposition added in this PR does not work with Global Phase. In addition, as Christina suggested, we should have a test catching the new error raised when using ctrl_zyz with more than one single control wire and a non-special unitary operator

Reply:
:100: reverted the test!